could not pass event NPCLeftClickEvent to Citizens dwarfcraft

Once a player clicks on a npc and this error pops up, it affects everyone until the server restarts

[00:22:21 ERROR]: Could not pass event NPCLeftClickEvent to Citizens v2.0.19-SNAPSHOT (build 1374)
org.bukkit.event.EventException
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.EventListen.onEntityDamage(EventListen.java:220) [Citizens.jar:?]
        at sun.reflect.GeneratedMethodAccessor40.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEvent(CraftEventFactory.java:89) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:572) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:479) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:604) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity0(EntityLiving.java:1238) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity0(EntityHuman.java:847) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity(EntityLiving.java:834) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity(EntityHuman.java:772) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.damageEntity(EntityPlayer.java:500) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.npc.entity.EntityHumanNPC.damageEntity(EntityHumanNPC.java:121) [Citizens.jar:?]
        at net.minecraft.server.v1_10_R1.EntityHuman.attack(EntityHuman.java:1009) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.attack(EntityPlayer.java:1136) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnection.a(PlayerConnection.java:1585) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:68) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:12) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnectionUtils$1.run(SourceFile:13) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_111]
        at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_111]
        at net.minecraft.server.v1_10_R1.SystemUtils.a(SourceFile:45) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.D(MinecraftServer.java:732) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.DedicatedServer.D(DedicatedServer.java:400) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.C(MinecraftServer.java:668) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.run(MinecraftServer.java:567) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
        at com.Jessy1237.DwarfCraft.DCPlayer.calculateTrainingCost(DCPlayer.java:67) ~[?:?]
        at com.Jessy1237.DwarfCraft.Out.printSkillInfo(Out.java:145) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.checkTrainerLeftClick(DCEntityListener.java:103) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.onNPCLeftClickEvent(DCEntityListener.java:543) ~[?:?]
        at com.Jessy1237.DwarfCraft.DwarfTrainerTrait.onNPCLeftClick(DwarfTrainerTrait.java:79) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor87.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        ... 35 more

Same for  Citizens v2.0.20-SNAPSHOT (build 1420). Also another thing, whenever a player destroys a block, it shoots up, not sure if there is a work around this. Btw thanks for the help.

Heres the error for 2.0.20 citizens:

[01:14:28 ERROR]: Could not pass event NPCLeftClickEvent to Citizens v2.0.20-SNAPSHOT (build 1420)
org.bukkit.event.EventException
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.EventListen.onEntityDamage(EventListen.java:221) [citizens-2.0.20-SNAPSHOT.jar:?]
        at sun.reflect.GeneratedMethodAccessor78.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEvent(CraftEventFactory.java:89) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:572) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:479) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:604) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity0(EntityLiving.java:1238) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity0(EntityHuman.java:847) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity(EntityLiving.java:834) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity(EntityHuman.java:772) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.damageEntity(EntityPlayer.java:500) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.nms.v1_10_R1.entity.EntityHumanNPC.damageEntity(EntityHumanNPC.java:122) [citizens-2.0.20-SNAPSHOT.jar:?]
        at net.minecraft.server.v1_10_R1.EntityHuman.attack(EntityHuman.java:1009) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.attack(EntityPlayer.java:1136) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnection.a(PlayerConnection.java:1585) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:68) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:12) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnectionUtils$1.run(SourceFile:13) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_111]
        at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_111]
        at net.minecraft.server.v1_10_R1.SystemUtils.a(SourceFile:45) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.D(MinecraftServer.java:732) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.DedicatedServer.D(DedicatedServer.java:400) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.C(MinecraftServer.java:668) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.run(MinecraftServer.java:567) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
        at com.Jessy1237.DwarfCraft.DCPlayer.calculateTrainingCost(DCPlayer.java:67) ~[?:?]
        at com.Jessy1237.DwarfCraft.Out.printSkillInfo(Out.java:145) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.checkTrainerLeftClick(DCEntityListener.java:103) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.onNPCLeftClickEvent(DCEntityListener.java:543) ~[?:?]
        at com.Jessy1237.DwarfCraft.DwarfTrainerTrait.onNPCLeftClick(DwarfTrainerTrait.java:79) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor57.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        ... 35 more
[01:14:43 WARN]: [DwarfCraft] Task #852 for DwarfCraft v3.1.1 generated an exception
java.lang.ArrayIndexOutOfBoundsException: 1
        at com.Jessy1237.DwarfCraft.DCPlayer.calculateTrainingCost(DCPlayer.java:67) ~[?:?]
        at com.Jessy1237.DwarfCraft.DwarfTrainer.trainSkill(DwarfTrainer.java:174) ~[?:?]
        at com.Jessy1237.DwarfCraft.TrainSkillSchedule.run(TrainSkillSchedule.java:18) ~[?:?]
        at org.bukkit.craftbukkit.v1_10_R1.scheduler.CraftTask.run(CraftTask.java:71) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:353) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.D(MinecraftServer.java:723) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.DedicatedServer.D(DedicatedServer.java:400) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.C(MinecraftServer.java:668) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.run(MinecraftServer.java:567) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]
[01:14:44 ERROR]: Could not pass event NPCLeftClickEvent to Citizens v2.0.20-SNAPSHOT (build 1420)
org.bukkit.event.EventException
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.EventListen.onEntityDamage(EventListen.java:221) [citizens-2.0.20-SNAPSHOT.jar:?]
        at sun.reflect.GeneratedMethodAccessor78.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEvent(CraftEventFactory.java:89) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:572) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:479) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:604) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity0(EntityLiving.java:1238) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity0(EntityHuman.java:847) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity(EntityLiving.java:834) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity(EntityHuman.java:772) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.damageEntity(EntityPlayer.java:500) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.nms.v1_10_R1.entity.EntityHumanNPC.damageEntity(EntityHumanNPC.java:122) [citizens-2.0.20-SNAPSHOT.jar:?]
        at net.minecraft.server.v1_10_R1.EntityHuman.attack(EntityHuman.java:1009) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.attack(EntityPlayer.java:1136) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnection.a(PlayerConnection.java:1585) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:68) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:12) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnectionUtils$1.run(SourceFile:13) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_111]
        at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_111]
        at net.minecraft.server.v1_10_R1.SystemUtils.a(SourceFile:45) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.D(MinecraftServer.java:732) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.DedicatedServer.D(DedicatedServer.java:400) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.C(MinecraftServer.java:668) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.run(MinecraftServer.java:567) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
        at com.Jessy1237.DwarfCraft.DCPlayer.calculateTrainingCost(DCPlayer.java:67) ~[?:?]
        at com.Jessy1237.DwarfCraft.Out.printSkillInfo(Out.java:145) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.checkTrainerLeftClick(DCEntityListener.java:103) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.onNPCLeftClickEvent(DCEntityListener.java:543) ~[?:?]
        at com.Jessy1237.DwarfCraft.DwarfTrainerTrait.onNPCLeftClick(DwarfTrainerTrait.java:79) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor57.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        ... 35 more
[01:14:46 ERROR]: Could not pass event NPCLeftClickEvent to Citizens v2.0.20-SNAPSHOT (build 1420)
org.bukkit.event.EventException
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.EventListen.onEntityDamage(EventListen.java:221) [citizens-2.0.20-SNAPSHOT.jar:?]
        at sun.reflect.GeneratedMethodAccessor78.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEvent(CraftEventFactory.java:89) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:572) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:479) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:604) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity0(EntityLiving.java:1238) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity0(EntityHuman.java:847) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity(EntityLiving.java:834) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity(EntityHuman.java:772) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.damageEntity(EntityPlayer.java:500) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.nms.v1_10_R1.entity.EntityHumanNPC.damageEntity(EntityHumanNPC.java:122) [citizens-2.0.20-SNAPSHOT.jar:?]
        at net.minecraft.server.v1_10_R1.EntityHuman.attack(EntityHuman.java:1009) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.attack(EntityPlayer.java:1136) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnection.a(PlayerConnection.java:1585) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:68) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:12) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnectionUtils$1.run(SourceFile:13) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_111]
        at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_111]
        at net.minecraft.server.v1_10_R1.SystemUtils.a(SourceFile:45) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.D(MinecraftServer.java:732) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.DedicatedServer.D(DedicatedServer.java:400) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.C(MinecraftServer.java:668) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.run(MinecraftServer.java:567) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
        at com.Jessy1237.DwarfCraft.DCPlayer.calculateTrainingCost(DCPlayer.java:67) ~[?:?]
        at com.Jessy1237.DwarfCraft.Out.printSkillInfo(Out.java:145) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.checkTrainerLeftClick(DCEntityListener.java:103) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.onNPCLeftClickEvent(DCEntityListener.java:543) ~[?:?]
        at com.Jessy1237.DwarfCraft.DwarfTrainerTrait.onNPCLeftClick(DwarfTrainerTrait.java:79) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor57.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        ... 35 more
[01:15:00 ERROR]: Could not pass event NPCLeftClickEvent to Citizens v2.0.20-SNAPSHOT (build 1420)
org.bukkit.event.EventException
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.EventListen.onEntityDamage(EventListen.java:221) [citizens-2.0.20-SNAPSHOT.jar:?]
        at sun.reflect.GeneratedMethodAccessor78.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEvent(CraftEventFactory.java:89) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:572) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:479) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:604) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity0(EntityLiving.java:1238) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity0(EntityHuman.java:847) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity(EntityLiving.java:834) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity(EntityHuman.java:772) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.damageEntity(EntityPlayer.java:500) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.nms.v1_10_R1.entity.EntityHumanNPC.damageEntity(EntityHumanNPC.java:122) [citizens-2.0.20-SNAPSHOT.jar:?]
        at net.minecraft.server.v1_10_R1.EntityHuman.attack(EntityHuman.java:1009) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.attack(EntityPlayer.java:1136) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnection.a(PlayerConnection.java:1585) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:68) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:12) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnectionUtils$1.run(SourceFile:13) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_111]
        at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_111]
        at net.minecraft.server.v1_10_R1.SystemUtils.a(SourceFile:45) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.D(MinecraftServer.java:732) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.DedicatedServer.D(DedicatedServer.java:400) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.C(MinecraftServer.java:668) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.run(MinecraftServer.java:567) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
        at com.Jessy1237.DwarfCraft.DCPlayer.calculateTrainingCost(DCPlayer.java:67) ~[?:?]
        at com.Jessy1237.DwarfCraft.Out.printSkillInfo(Out.java:145) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.checkTrainerLeftClick(DCEntityListener.java:103) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.onNPCLeftClickEvent(DCEntityListener.java:543) ~[?:?]
        at com.Jessy1237.DwarfCraft.DwarfTrainerTrait.onNPCLeftClick(DwarfTrainerTrait.java:79) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor57.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        ... 35 more
[01:15:01 WARN]: [DwarfCraft] Task #856 for DwarfCraft v3.1.1 generated an exception
java.lang.ArrayIndexOutOfBoundsException: 1
        at com.Jessy1237.DwarfCraft.DCPlayer.calculateTrainingCost(DCPlayer.java:67) ~[?:?]
        at com.Jessy1237.DwarfCraft.DwarfTrainer.trainSkill(DwarfTrainer.java:174) ~[?:?]
        at com.Jessy1237.DwarfCraft.TrainSkillSchedule.run(TrainSkillSchedule.java:18) ~[?:?]
        at org.bukkit.craftbukkit.v1_10_R1.scheduler.CraftTask.run(CraftTask.java:71) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:353) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.D(MinecraftServer.java:723) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.DedicatedServer.D(DedicatedServer.java:400) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.C(MinecraftServer.java:668) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.run(MinecraftServer.java:567) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]
[01:15:03 ERROR]: Could not pass event NPCLeftClickEvent to Citizens v2.0.20-SNAPSHOT (build 1420)
org.bukkit.event.EventException
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.EventListen.onEntityDamage(EventListen.java:221) [citizens-2.0.20-SNAPSHOT.jar:?]
        at sun.reflect.GeneratedMethodAccessor78.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEvent(CraftEventFactory.java:89) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:572) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:479) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:604) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity0(EntityLiving.java:1238) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity0(EntityHuman.java:847) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity(EntityLiving.java:834) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity(EntityHuman.java:772) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.damageEntity(EntityPlayer.java:500) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.nms.v1_10_R1.entity.EntityHumanNPC.damageEntity(EntityHumanNPC.java:122) [citizens-2.0.20-SNAPSHOT.jar:?]
        at net.minecraft.server.v1_10_R1.EntityHuman.attack(EntityHuman.java:1009) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.attack(EntityPlayer.java:1136) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnection.a(PlayerConnection.java:1585) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:68) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:12) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnectionUtils$1.run(SourceFile:13) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_111]
        at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_111]
        at net.minecraft.server.v1_10_R1.SystemUtils.a(SourceFile:45) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.D(MinecraftServer.java:732) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.DedicatedServer.D(DedicatedServer.java:400) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.C(MinecraftServer.java:668) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.run(MinecraftServer.java:567) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
        at com.Jessy1237.DwarfCraft.DCPlayer.calculateTrainingCost(DCPlayer.java:67) ~[?:?]
        at com.Jessy1237.DwarfCraft.Out.printSkillInfo(Out.java:145) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.checkTrainerLeftClick(DCEntityListener.java:103) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.onNPCLeftClickEvent(DCEntityListener.java:543) ~[?:?]
        at com.Jessy1237.DwarfCraft.DwarfTrainerTrait.onNPCLeftClick(DwarfTrainerTrait.java:79) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor57.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        ... 35 more
[01:15:07 ERROR]: Could not pass event NPCLeftClickEvent to Citizens v2.0.20-SNAPSHOT (build 1420)
org.bukkit.event.EventException
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.EventListen.onEntityDamage(EventListen.java:221) [citizens-2.0.20-SNAPSHOT.jar:?]
        at sun.reflect.GeneratedMethodAccessor78.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEvent(CraftEventFactory.java:89) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:572) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:479) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:604) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity0(EntityLiving.java:1238) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity0(EntityHuman.java:847) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityLiving.damageEntity(EntityLiving.java:834) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityHuman.damageEntity(EntityHuman.java:772) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.damageEntity(EntityPlayer.java:500) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.citizensnpcs.nms.v1_10_R1.entity.EntityHumanNPC.damageEntity(EntityHumanNPC.java:122) [citizens-2.0.20-SNAPSHOT.jar:?]
        at net.minecraft.server.v1_10_R1.EntityHuman.attack(EntityHuman.java:1009) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.EntityPlayer.attack(EntityPlayer.java:1136) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnection.a(PlayerConnection.java:1585) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:68) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PacketPlayInUseEntity.a(SourceFile:12) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.PlayerConnectionUtils$1.run(SourceFile:13) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_111]
        at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_111]
        at net.minecraft.server.v1_10_R1.SystemUtils.a(SourceFile:45) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.D(MinecraftServer.java:732) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.DedicatedServer.D(DedicatedServer.java:400) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.C(MinecraftServer.java:668) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.run(MinecraftServer.java:567) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
        at com.Jessy1237.DwarfCraft.DCPlayer.calculateTrainingCost(DCPlayer.java:67) ~[?:?]
        at com.Jessy1237.DwarfCraft.Out.printSkillInfo(Out.java:145) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.checkTrainerLeftClick(DCEntityListener.java:103) ~[?:?]
        at com.Jessy1237.DwarfCraft.listeners.DCEntityListener.onNPCLeftClickEvent(DCEntityListener.java:543) ~[?:?]
        at com.Jessy1237.DwarfCraft.DwarfTrainerTrait.onNPCLeftClick(DwarfTrainerTrait.java:79) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor57.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_111]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_111]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        ... 35 more
[01:15:08 WARN]: [DwarfCraft] Task #859 for DwarfCraft v3.1.1 generated an exception
java.lang.ArrayIndexOutOfBoundsException: 1
        at com.Jessy1237.DwarfCraft.DCPlayer.calculateTrainingCost(DCPlayer.java:67) ~[?:?]
        at com.Jessy1237.DwarfCraft.DwarfTrainer.trainSkill(DwarfTrainer.java:174) ~[?:?]
        at com.Jessy1237.DwarfCraft.TrainSkillSchedule.run(TrainSkillSchedule.java:18) ~[?:?]
        at org.bukkit.craftbukkit.v1_10_R1.scheduler.CraftTask.run(CraftTask.java:71) ~[spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at org.bukkit.craftbukkit.v1_10_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:353) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.D(MinecraftServer.java:723) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.DedicatedServer.D(DedicatedServer.java:400) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.C(MinecraftServer.java:668) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at net.minecraft.server.v1_10_R1.MinecraftServer.run(MinecraftServer.java:567) [spigot-1.10.2.jar:git-Spigot-570526c-c1b529e]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_111]

--- a/plugin.yml
+++ b/plugin.yml
@@ -2,7 +2,7 @@ name: DwarfCraft
 description: A skills and progression plugin for character development in RPG worlds.
 author: Jessy1237
 main: com.Jessy1237.DwarfCraft.DwarfCraft
-version: 3.1.1
+version: 3.1.2
 depend: [Vault, Citizens]
 softdepend: [LogBlock]
 commands: 

--- a/pom.xml
+++ b/pom.xml
@@ -5,7 +5,7 @@
     <artifactId>DwarfCraft</artifactId>
     <name>DwarfCraft</name>
     <packaging>jar</packaging>
-    <version>3.1.1</version>
+    <version>3.1.2</version>
 
     <repositories>
         <repository>

--- a/src/com/Jessy1237/DwarfCraft/DCPlayer.java
+++ b/src/com/Jessy1237/DwarfCraft/DCPlayer.java
@@ -62,7 +62,7 @@ public class DCPlayer
             System.out.println( "DC0: starting skill ordering for quartiles" );
         for ( Skill s : getSkills().values() )
         {
-            if ( s.getLevel() > 5 )
+            if ( s.getLevel() > plugin.getConfigManager().getRaceLevelLimit() )
             {
                 levelList[i] = s.getLevel();
                 i++;

--- a/src/com/Jessy1237/DwarfCraft/Out.java
+++ b/src/com/Jessy1237/DwarfCraft/Out.java
@@ -117,7 +117,7 @@ public class Out
     public boolean printSkillInfo( CommandSender sender, Skill skill, DCPlayer dCPlayer, int maxTrainLevel )
     {
         // general line
-        sendMessage( sender, Messages.skillInfoHeader.replaceAll( "%playername%", dCPlayer.getPlayer().getDisplayName() ).replaceAll( "%skillid%", "" + skill.getId() ).replaceAll( "%skilllevel%", "" + skill.getLevel() )
+        sendMessage( sender, Messages.skillInfoHeader.replaceAll( "%playername%", dCPlayer.getPlayer().getDisplayName() ).replaceAll( "%skillid%", "" + skill.getId() ).replaceAll( "%skillname%", "" + skill.getDisplayName() ).replaceAll( "%skilllevel%", "" + skill.getLevel() )
                 .replaceAll( "%maxskilllevel%", "" + plugin.getConfigManager().getMaxSkillLevel() ) );
 
         // effects lines
@@ -330,8 +330,7 @@ public class Out
     /**
      * Sends a welcome message based on race of player joining. Broadcasts to
      * the whole server
-     * 
-     * @param server
+     *
      * @param dCPlayer
      */
     public void welcome( DCPlayer dCPlayer )

